Allow skipping of production tags from tests

pauljamescleary opened the pull request:
Some of our func tests are not meant to be run against "real" environments.  Since we now run the functest container against non docker environments, we have to skip these tests.

mitruly commented on docker/functest/run.sh:
```diff
@@ -15,6 +15,13 @@ else
+if [ -z "${FOR_PROD}" ]; then
+  SKIP_TAGS=
+else
+  SKIP_TAGS="-m \"not skip_production\""
```
Since `SKIP_TAGS` would be a set value (rather than passing something in), it feels weird to use `-z` with an env var. Would prefer to see a parameter passed into `run.sh` (eg. `-s`/`--skip-production`) which would set the command.

pauljamescleary replied:
We have to get the value in from the `Dockerfile`.  ENV vars are the easiest for that

nimaeskandary commented:
Hm getting a "vinyldns-functest | collecting ... ERROR: file not found: skip_production" when trying this out locally, investigating

britneywright commented on docker/functest/run.sh:
```diff
@@ -15,6 +15,13 @@ else
   TEST_PATTERN="-k ${TEST_PATTERN}"
 fi
 
+# If we are to skip tests not suitable for prod environments, set to true or omit
```
```suggestion
# Assume we are not in a production environment unless this is set. Suggest setting to true if in a production environment. 
```

britneywright commented on docker/functest/run.sh:
```diff
@@ -15,6 +15,13 @@ else
 fi
 
+# If we are to skip tests not suitable for prod environments, set to true or omit
+if [ -z "${FOR_PROD}" ]; then
```
can we rename this to PROD or PROD_ENV

pauljamescleary commented on docker/functest/run.sh:
```diff
@@ -42,4 +49,4 @@ done
 
 cd /app
-./run-tests.py live_tests -v ${TEST_PATTERN} --url=${VINYLDNS_URL} --dns-ip=${DNS_IP}
+./run-tests.py live_tests -v ${TEST_PATTERN} ${SKIP_TAGS} --url=${VINYLDNS_URL} --dns-ip=${DNS_IP}
```
cool, thanks

britneywright commented on docker/functest/run.sh:
```diff
@@ -42,4 +42,13 @@ done
+# If PROD_ENV is unset, we are in a local docker environment so do not skip anything
+if [ -z "${PROD_ENV}" ]; then
+    echo "./run-tests.py live_tests -v --url=${VINYLDNS_URL} --dns-ip=${DNS_IP} ${TEST_PATTERN}"
+    ./run-tests.py live_tests -v --url=${VINYLDNS_URL} --dns-ip=${DNS_IP} ${TEST_PATTERN}
```
the "not skip_production" should be here right, since this is the prod environment?

pauljamescleary replied:
`-z` means "if not"

britneywright replied:
thanks :)

britneywright commented on docker/functest/run.sh:
```diff
@@ -42,4 +42,13 @@ done
+    ./run-tests.py live_tests -v --url=${VINYLDNS_URL} --dns-ip=${DNS_IP} ${TEST_PATTERN}
+else
+    # -m plays havoc with -k, using variables is a headache, so doing this by hand
+    echo "./run-tests.py live_tests -m \"not skip_production\" -v --url=${VINYLDNS_URL} --dns-ip=${DNS_IP} ${TEST_PATTERN}"
```
and "not skip_production" shouldn't be here because this is the local environment.

pauljamescleary replied:
yea, if we are not in the prod environment, then run everything.

Otherwise, we are not in a prod environment, so skip certain tests.

nimaeskandary commented on docker/functest/run.sh:
```diff
@@ -42,4 +42,13 @@ done
-./run-tests.py live_tests -v ${TEST_PATTERN} --url=${VINYLDNS_URL} --dns-ip=${DNS_IP}
+
+# If PROD_ENV is unset, we are in a local docker environment so do not skip anything
+if [ -z "${PROD_ENV}" ]; then
```
```suggestion
if [ "${PROD_ENV}" != "true"]; then
```

nimaeskandary replied:
Could be confusing because if someone sets PROD_ENV to false it would still treat it the same

britneywright commented on docker/functest/run.sh:
```diff
@@ -42,4 +42,13 @@ done
 cd /app
-./run-tests.py live_tests -v ${TEST_PATTERN} --url=${VINYLDNS_URL} --dns-ip=${DNS_IP}
+
+# If PROD_ENV is unset, we are in a local docker environment so do not skip anything
```
comment needs to be updated. Should be something like "if PROD_ENV is unset or not set to true".

pauljamescleary commented:
@rebstar6 I updated, basically if the PROD_ENV was unset, that first != was still firing.

I switched up the order.  You _HAVE_ to set PROD_ENV=true for that to be skipped now.
